api/turing/config/config.go

```diff
@@ -285,10 +285,18 @@ type AuthorizationConfig struct {
 	URL     string
 }
 
+type ClusterConfig struct {
+	// InClusterConfig is a flag if the service account is provided in Kubernetes
+	// and has the relevant credentials to handle all cluster operations.
+	InClusterConfig bool
+	// VaultConfig is required if InClusterConfig is false.
+	VaultConfig *VaultConfig
+}
+
 // VaultConfig captures the config for connecting to the Vault server
 type VaultConfig struct {
-	Address string `validate:"required"`
-	Token   string `validate:"required"`
```

Comment:
Maybe it still is `required`, but `VaultConfig` itself is not `required` anymore. I.e. `VaultConfig` (line 293) can be `nil`, but if it's not nil, then `Address`/`Token` are `required`